core/deployment/src/main/java/io/quarkus/deployment/steps/RegisterForReflectionBuildStep.java

```diff
@@ -30,18 +33,48 @@
     public void build() throws Exception {
         for (AnnotationInstance i : combinedIndexBuildItem.getIndex()
                 .getAnnotations(DotName.createSimple(RegisterForReflection.class.getName()))) {
-            ClassInfo target = i.target().asClass();
+
             boolean methods = i.value("methods") == null || i.value("methods").asBoolean();
             boolean fields = i.value("fields") == null || i.value("fields").asBoolean();
+
             AnnotationValue targetsValue = i.value("targets");
-            if (targetsValue == null) {
+            AnnotationValue namesValue = i.value("names");
+
+            if (targetsValue == null && namesValue == null) {
+                ClassInfo target = i.target().asClass();
                 reflectiveClass.produce(new ReflectiveClassBuildItem(methods, fields, target.name().toString()));
-            } else {
+                return;
+            }
+
+            Set<String> classNames = new HashSet<>();
+            if (targetsValue != null) {
                 Type[] targets = targetsValue.asClassArray();
                 for (Type type : targets) {
-                    reflectiveClass.produce(new ReflectiveClassBuildItem(methods, fields, type.name().toString()));
+                    classNames.add(String.valueOf(type.name()));
+                }
+            }
+
+            if (namesValue != null) {
+                String[] names = namesValue.asStringArray();
+                for (String name : names) {
+                    if (isClass(name)) {
+                        classNames.add(name);
+                    }
                 }
             }
+
+            for (String className : classNames) {
+                reflectiveClass.produce(new ReflectiveClassBuildItem(methods, fields, className));
+            }
+        }
+    }
+
+    private boolean isClass(String className) {
+        try {
+            Class.forName(className, false, getClass().getClassLoader());
+            return true;
+        } catch (ClassNotFoundException e) {
+            return false;
```

Comment:
If we are going to load classes (I assume to make things safer for users, we should at least be logging when the load failed.

But as I said above, I don't really see much point in this type of validation)

Reply:
Removed the Validation, This won't be required now.